packages/transaction-controller/src/TransactionController.ts

```diff
@@ -646,22 +624,111 @@ export class TransactionController extends BaseControllerV1<
     this.messagingSystem.subscribe(
       'NetworkController:stateChange',
       (_, patches) => {
-        if (this.#isMultichainEnabled) {
-          const networkClients = this.#getNetworkClientRegistry();
-          patches.forEach(({ op, path }) => {
-            if (op === 'remove' && path[0] === 'networkConfigurations') {
-              const networkClientId = path[1] as NetworkClientId;
-              delete networkClients[networkClientId];
-            }
-          });
+        // only needed to get `should not call getNetworkClientRegistry on networkController:stateChange when feature flag is disabled` to pass,
+        // otherwise can rely on the guard inside the helper
+        if (isMultichainEnabled) {
```

Comment:
This is the only usage of isMultichainEnabled left in the controller after this refactor. I've left it here so that I don't need to update one of the integration tests that rely on it.